### ISSUE TITLE
Game Music

### ISSUE THREAD

**Issue by BearbaBear** [endless-sky/endless-sky#3594]
If this game had music, what would you like it to be?  I am a musician, and may be able to write something.

**Comment by KiLEdEnNis** [comment]
~~Pendulum~~
this has been brought up before, I'll link ya to the issues and some important things MZ (the creator) said
#44:
> But right now in-game music seems like a low priority to me - it'd have to be really high quality music in order to convince the player to not just disable it and provide their own music via an MP3 player in the background instead.

#233:
> he only way it would be worth adding music to the game would be if the music were so good that people would not just turn it off and play music that's more to their own liking in the background, instead. (At least, that's what I do with almost every game I play.) So that sets a really high bar - the music needs to be able to compete with professional recordings, and has to be in a genre that players like.

> It really confuses me why people don't just play music of their own choosing while playing games. There are very few games that have music good enough that I'd listen to it over my own mp3 collection.
I do plan to add a "music" engine to the game eventually, but the first thing it will be used for will be to add ambient sounds when you're landed on a planet, to add a bit more atmosphere - e.g. ocean waves on one planet, wind on another, etc.

basically, it's a low-priority thing, but it was planned
if you can create some music that most players would like it'd be greatly appreciated, but the bar *is* high
either way, good luck, and if ya want ya can also make sounds for the game

**Comment by BearbaBear** [comment]
Thank you for that information.  One problem is that I don't know what kinds of music most players would like.  I'm thinking something synthy and a bit 80s, but I don't know what would be most appreciated by players.

**Comment by warp-core** [comment]
The music should suit the game. If it does, then the kind of people that like the game would likely like the music. Maybe just try some things and see how well it fits?

**Comment by kikotheexile** [comment]
Look at music from similar settings across genres. Synth- orchestral is a pretty common mood setting theme. Take some trans Siberian and smash it together with some daft punk discovery-esque sounds and you might start finding something that gives you some real inspiration.

**Comment by kikotheexile** [comment]
For reference, this is the sort of thing i listen to while playing ES. Though the lyrics can get in the way of the mood. This is like a tevh heavy metal with orchestral elements.

https://youtu.be/uozt4N72a5U?list=PLDp_PtuyOwCYS8mEf7sImnsoSiALQB9R9

**Comment by MCOfficer** [comment]
i was working on this for a while, but like MZ says, there's a high bar - at some point i realized it couldn't live up, say, the FTL soundtrack or something similiar.

since we're doing song suggestions here, lemme recommend widek's album "journey to the stars".

**Comment by KiLEdEnNis** [comment]
~~all 3 Pendulum albums~~
apart from the lyrics, I think something similar to [this](https://www.youtube.com/watch?v=6EKYcAS_2ys) would fit reasonably well in the game

but how should "we" start though, have BearbaBear create some samples, or agree on a general theme, genre or style beforehand? as it stands now I think it's just waiting on samples

**Comment by Amazinite** [comment]
I imagine music found on a channel like [Electronic Gems](https://www.youtube.com/user/HungOverGargoyle/videos) could work well; the music there generally has a more retro sound to it, something that I feel fits with the game when I'm playing. I'd imagine though that music tailored for the game would be fairly minimalistic, nothing that's designed to pop out at the player when they're trying to focus on the game but just enough that it's something they'll enjoy listening to while just floating through space or hyperspacing between systems without being in the middle of combat. 

I think there's also been talk about music changing depending on the situation, such as different music when you're in combat, possibly even changing depending on the faction you're fighting at the moment. Combat music could spare to have a little more pop to it, as the player will be focusing on the combat more than the music anyways.

Different music for different regions of space would be interesting too. Something like [this](https://youtu.be/KbhSHdo8eOA) just feels so Paradise Worlds to me, but something like [this](https://www.youtube.com/watch?v=BSC6d81pvwE) feels like it would fit more for the Rim, or [this](https://www.youtube.com/watch?v=oxoqm05c7yA) for the Core. (I'm not really sure how to explain each of those feelings, though.) Different races could even use different genres of music. Human space could take heavily from more synth heavy genres like I've been linking, but Korath space could have more rock stuff like @kikotheexile linked (what he specifically linked sounding a lot like Korath combat music), and Wanderer space could have wind instrument and/or string instrument music, something like [this](https://youtu.be/DhHGDOgjie4).

**Comment by juanminm** [comment]
@Amazinite I was thinking the same about different music for different parts, or even lack of it. The systems where the Drak have destroyed their star feels like something where there would be either no BGM or an ambient sound that would represent what the captain felt on that Drak system: "the words are accompanied by an emotion that you cannot put a name to" and "the words "prevent extinction" echo loud in your mind as if underlined by a nameless emotion".

**Comment by endless-sky** [comment]
I mostly listen to acoustic / folk music while playing, because [Firefly](https://en.wikipedia.org/wiki/Firefly_(TV_series)) has convinced me that outer space is better with banjos.

**Comment by solardawning** [comment]
To chime in with the other perspective:

When I play games, I don't listen to music of my own- ever. I very much appreciate background music in games as a part of the creator's overall vision and intent. Game music helps to set the scene for the player, adjust their mood, and help them feel at ease- or not, with the game world, as the designer chooses.

When playing Endless Sky, I don't listen to music, because that would take me too much 'out' of the game.

**Comment by BearbaBear** [comment]
I see some great ideas here.  I especially like the different genres for different races, even though I am not in a position to record a hard rock song, for instance.

Now, I did some thinking, and I came to the conclusion that something very atmospheric and subtle would be the best fit.  Even with pirates, the game has a very laid-back feel to it, and this type of music would help shape the mood of it.  It could be made uneasy/tense with a few harmonic tricks, maybe for use in battle, but still able to keep its same atmospheric qualities.  

That type of music is also generally hard to judge.  I agree with @solardawning wholeheartedly.  Again, I honestly don't feel like music of an atmospheric nature would be easy to judge as "good" or "bad" or "amazing", for instance, mainly because of its very subtle nature.  It would serve its purpose, and well, but I don't think it quite needs to reach the sky-high standards that I have seen being circulated.

**Comment by Pointedstick** [comment]
@endless-sky I do the same thing! [Brown Bird](https://brownbird.bandcamp.com/) is pretty great Firefly-style acoustic folk music for playing Endless Sky to, if I can make a recommendation.

**Comment by geojak** [comment]
it would be really cool if there was a music engine so I can add the EV intro songs to play whenever I launch ES. 

Other than that, if there was a dynamic system which picks songs from categories like
battle music
almost dead intense music
menu music
launch music
music in syndicate space
music in republic space
music in hai space
custom
etc..
it would add alot of character to each region and faction.
and new mp3-player button that plays songs from the custom folder the players can add music to.
or just turn music off and play as is.

I personally dont like to play my own music as background and prefer the fitting soundtracks in most games.

would need the engine for dynamic music first and decide on categories. people will surely start contributing songs then.

**Comment by jostephd** [comment]
Mission authors could be able to specify custom categories (or outright custom tracks) for their missions.

**Comment by jostephd** [comment]
Wild idea: Danforth could say "Good Luck" (in voice) as you're lining up to jump into the final battle. (That message could also be displayed in the messages ticker at the bottom of the playfield at the same time)

**Comment by RizkyPramudyaCJ** [comment]
For me, the music for ES is might be :

1. Faction Music
2. Mission Music
3. Battle Music
4. Victory after main mission music.
5. Normal Roaming music.
6. Death music.

Thats a music which should be added on ES.

**Comment by brendanjones** [comment]
> Game music helps to set the scene for the player, adjust their mood, and help them feel at ease- or not, with the game world, as the designer chooses.

This. I can still whistle the intro music to every Escape Velocity game, and feel my 10-yr-old self's excitement and the impending space exploration/explosions. 

For the uninitiated, here's the intro music for:
Original EV: https://www.youtube.com/watch?v=cnBP7GwPw-U
Override: https://www.youtube.com/watch?v=WFIvLPF__L4
Nova: https://www.youtube.com/watch?v=nG-i6MSnuFQ

I generally don't like music in-game because it's a distraction from game-play, but that's because my normal gaming fare is intensive combat-fests like DotA where audio effects can be just as important to snap decisions as visual cues. ES is a lot more chill, and any time not spent in combat I'd happily have supplemented by mood music.

I say just use the Stargate theme music for the intro: https://www.youtube.com/watch?v=MnFQlVT2UBo :D

**Comment by Tadrix** [comment]
Just a heads-up: The EV Nova intro music sounds like [Gustav Holst's](https://en.wikipedia.org/wiki/Gustav_Holst) [_Mars, the Bringer of War_](https://en.wikipedia.org/wiki/File:Holst-_mars.ogg), from [_The Planets_](https://musopen.org/music/43775-the-planets-op-32/), written in 1914. It's often used in various soundtracks. Gustav Holst lived from 1874 till 1934, therefore his music is in the public domain.

**Comment by brendanjones** [comment]
Wow, never knew that! Nice spot.

**Comment by Zitchas** [comment]
Having dynamic music would be a really nice addition. I, too, still remember the trilogy intro tunes. (although to be honest, I'm thrilled that ES loads *so* much faster than any of the trilogy, so it doesn't really need an intro piece to keep my attention.)

**Comment by Pointedstick** [comment]
Mass Effect 1 had IMHO near-perfect space opera genre music.  That music struck just the right tone and feel for a game like it, or Endless Sky.

**Comment by NCGThompson** [comment]
Since it's open world the game can have a few different "songs" it can play in between times of quite. Different conditions, like locations can have their own tracks that the game mixes together as they change. For example, an exotic Si-Fi instrument can fade in once the player enters Quarg space.
That being said it is possible we may want events to interrupt the "song".
*That* being said combat is going to be tricky.As @BearbaBear said there is a difference between when a hostile ships targets the player and when the player is actually going to be mentally engaged in combat. The current combat alarm eventually becomes the "Welcome to this system!" alarm. My idea is we came the alarm for utilitarian purposes but the game still waits to see if the player will stay in combat then gradually change the music or wait until a transition point in the music.
Nonetheless someone has to get started on the engine.

Unrelated unpopular opinion: Retro music is cliché.

**Comment by BearbaBear** [comment]
Is anyone familiar with the game SUPERHOT? If you aren't (or you are and you just didn't notice this), it contains a soundtrack that most people don't seem to ever realize is there.  It definitely is there, though, and if it wasn't, people would notice.  It simply exists to fill the space.  Endless Sky has nothing, and that's why (to me, at least) its audio always sounds so empty.  And yes, space _is_ supposed to sound empty, but I think adding a bit of background ambiance would make things sound so much more natural.  https://soundcloud.com/the-fummies/sets/superhot-soundtrack

I'm not even thinking about music anymore.  Now I'm just thinking about how to make background noise that sounds like it could be in space. If sky-high standards are expected then it's gonna take a while to make a worthy soundtrack, and I wouldn't want ES to sound so empty for so long. Background noise would be pretty simple to implement, since it wouldn't require accounting for combat situations or anything.  Until a soundtrack is completed, this would be best for the game.

**Comment by NCGThompson** [comment]
A technical consideration: Currently the whole game is loaded to memory at launch and the only other times the it loads from disk is changing save files. Is loading assets (such as music) asynchronously out of the question? And if we can do that, what is the minimum hard drive read speed we can expect?

One thing we need to make a sound track is know how big the music files be.

**Comment by MCOfficer** [comment]
i guess music would be stored as mp4 (like the ambient sounds). that's a few MB for a 3 minute-track (can vary greatly depening on quality), so with over 50MB of images we probably don't need to consider that.

**Comment by BearbaBear** [comment]
Could even just compress it by an ungodly amount and call it an "artistic decision"

**Comment by Zitchas** [comment]
On my system Endless Sky uses 358Mb initially, then dropped to 324Mb when I was wandering around hunting Korath with 30+ Shield Beetles. 

Given that the standard build for computers these days have a bare minimum of 4Gb, with 6 or 8 GB being quite common, I don't think having an extra 50Mb of data loaded at startup is going to cause any hardship for anyone. We shouldn't bloat the game just for the sake of doing so, but I think 50Mb of extra memory usage in order to have 45-60min of music would be quite reasonable. (For comparison, my 12 year old laptop that is so old it can't run ES has 4 Gb of RAM in it. And laptop standard specs have improved at least a little since then...)

The only caveat I can think of is having it set up so that if the user unchecks the music option (or sets the volume to 0), then the music files aren't loaded. This would give users running on edge cases where the difference between 350Mb and 400Mb of RAM usage is actually critical the ability to save that space.

**Comment by endless-sky** [comment]
The game is already designed to stream MP3s from disk rather than loading all the music files into memory. Given the bitrate of an MP3 and the speed of a hard drive, it's unlikely to stall unless there's an issue with the disk.

**Comment by MCOfficer** [comment]
I think at this point we've established that 

a) Situation based music is really damn hard.
b) none of us is likely to create a universally accepted soundtrack, in part because
c) People prefer different kinds of music while playing ES.

---

So what about this: Have a an outfit that basically functions as radio. The player can choose between channels, and each channel randomly plays several selected songs of similar style (one banjo channel, one for ambiance, etc.).

1. Such a system would be very customizable. Adding a new channel is as simple as creating a small plug-in with a few lines of text and some music files. (Would work best with #707 for ease of use)

2. It solves c) very elegantly, by just giving the choice to the player.

3. It even makes some sense within the game itself. We already know there are hyperspace beacons, so there might as well be radio stations using the same technology.

4. However, it'd also throw the idea of situation-based music out of the window.


Additionally we could have some cool effects, like have the nano bot swarm transmit a creepy-crawly loop that is layered over the music.

**Comment by BearbaBear** [comment]
That's actually a great idea. Things like radio hijackers would be cool for pirates to have on their ships. In regards to the music, would it be only user-set music, or would the game ship with original music in different styles?

**Comment by MCOfficer** [comment]
We'd definitely need some "stock" channels. We can't force players to download plug-ins just to use a functionality that is part of the game.

**Comment by jafdy** [comment]
For steam you could have them as free DLC (like the high DPI plugin).
Also consider having a way to listen to internet radio, in the same vain as PC Buliding Simulator. Then people can listen to whatever internet radio they want (as long as they have internet) and it doesn't increases file size, stays up-to-date and requires very little input beyond implementation. Although I imagine that implementing that would be hard.

**Comment by Amazinite** [comment]
The idea seems interesting (especially tying it into the hyperspace relays, ignoring the fact that the player isn't going to always been connected to them), but at that point how is it any different than just already using your own music, albeit with some "suggestions" from the game itself. ~~If I can't listen to "Sad Archie" then what's the point?~~

**Comment by BearbaBear** [comment]
If an in-game radio is added, I think it would absolutely need to be able to play user-added music. This would add convenience for people who already do that and would give the game an option to interact with the music as necessary, such as controlling its volume or distorting it as certain events could require.

Beyond internet radio, which I think is extremely restrictive, direct Spotify integration is definitely possible and could allow players to choose their own playlists. This would also be great for event-based music, as Spotify keeps metadata for every one of their tracks that states their "energy level" and similar statistics. While these numbers are algorithmically determined, I have found that they are surprisingly accurate in their judgement. The game could use such stats for songs in the user-made playlists, automatically playing songs from them with certain energy levels and moods at certain times as events could request.

**Comment by MCOfficer** [comment]
ok, hold your horses here. internet radio and spotify integration are cool ideas, but as long as we can't get an integrated plug-in manager going, i wouldn't get my hopes up for this.

I got this inspiration from another space game, oolite (an elite clone), which doesn't have music by default. However, there is an add-on that does exactly what i suggested - and it's being used. So apparently there *is* a difference between having a radio outfit and having spotify running.

**Comment by jafdy** [comment]
Spotify almost certainly has a closed API, and if so it is effectively impossible for use to use due to licensing issues.

**Comment by Cerebrate-Daggoth** [comment]
If a radio type thing happens to come to life with various channels of differing music, i'd recommend a cheesy elevator music channel for the heck of it. if not, no worries. just think it'd be a funny thing to have.

**Comment by Zitchas** [comment]
There's a lot of different music services out there. There's got to be one that is open software friendly, right?

**Comment by tehhowch** [comment]
Not gonna add netcode just for music streaming when ES is resource-friendly enough that you can run it in a tab or desktop app without issue. We'll need netcode for an online plugin service, so after that, one could reasonably expect that a connection elsewhere would be feasible. But, before an online plugin service comes an offline one.

Also not sure I want to need to worry about saving user credentials (e.g. your Spotify account stays logged in) or manage third party API credentials securely for an otherwise offline (no central ES server to do the authentication/session initiation & client handoff) game.

In other words, a radio outfit that plays only local content is gonna happen long before we add music streaming. (Unless of course someone PRs a sane implementation.)

**Comment by AlbertNewton** [comment]
Looked through, and can't find this anywhere else.
I need to put this here. @Ferociousfeind and I have been working on it
https://docs.google.com/document/d/1Tayd8HEXBMD3-LFOOY5RWAwatDsw-XhRCHiB3crSMk4/edit

**Comment by silvernode** [comment]
> Spotify almost certainly has a closed API, and if so it is effectively impossible for use to use due to licensing issues.

What about the Pandora API? There are already free applications that use this like Pianobar and Pithos. I think Pithos just uses the pianobar functions which are written in C.

**Comment by dfrankow** [comment]
What's the state of this? Are there hooks someone can use to drop in music or background audio?

**Comment by MCOfficer** [comment]
technically, there is code for sounds which are played once you enter a system, and there are sounds for planets (which are currently used for ambiance). none of those could act as a substitute for what this issue is about, though.

**Comment by dfrankow** [comment]
When I tried the 0.9.10, it's silent all the time. Should I open up another issue about getting some hooks for music to where people could contribute easily? Or maybe that's in the wiki somewhere?

**Comment by AlbertNewton** [comment]
I tried to access the former system back in 0.9.9, @MCOfficer 
I couldn't get it to play a sound when I entered a system

**Comment by MCOfficer** [comment]
> 
> 
> I tried to access the former system back in 0.9.9, @MCOfficer
> I couldn't get it to play a sound when I entered a system

i don't know if or how it works - just saying the code is there.

**Comment by AlbertNewton** [comment]
I can confirm that. I don't speak C++, but the name and contents of music.cpp are relatively self-explanatory.

**Comment by Amazinite** [comment]
Ambient sounds are currently only used on stations. https://github.com/endless-sky/endless-sky/commit/6d913471705e982e34d73a7d28399f3c4dab94a7
All systems and planets are able to play music. https://github.com/endless-sky/endless-sky/commit/bb3f92238b48603e433532e9710f2123814196d1
The ultimate intention though is that the music change to match the situation (e.g. if in combat or not).

**Comment by AlbertNewton** [comment]
To sum up my understanding(I used to be involved with composing some music to be implemented into the game, a project which sadly seems since to have fallen by the wayside), we need music to vary:
1: According to the current player position. Are they docked or in space? Music and ambient noise will differ for each.
2: Where in space are they? Pirate space is going to have very different music to landed on a planet colonised primarily by the Kimek.
3: According to situation. Music needs to change when combat starts or ends, to name one example.

Ideally, we would have a system that can look at all aspects of the game situation and "compose" its own music. For example, when the player enters a Free Worlds system bordering a pirate system, Free Worlds music will play(or continue to play). When pirate ships enter the system, motifs that can be heard in pirate space start to appear in the music. When they target the player, the motifs become more and more prominent, until some Navy ships warp in. Now, the soundscape is a three-way battle between the FW, the pirates and the Navy, and the relative weights of each style of music changes as each force seems to be winning or losing the fight.
The player gets hit by a few heavy missiles, and is destroyed. The music becomes more sombre, then fades into nothing but a faint white noise.
Now, this would be awesome, but it'd be hard to build, and near impossible to make sound good, and it'd probably take a ton of system resources.

Even the system discussed at the start of this comment would be very difficult to make well. I know; I tried to do it myself(though it should be noted that I never actually tried to grapple with the game code, merely tried to pseudocode some algorithms to decide which track to play). It seemed important that tracks shouldn't switch around too often, because otherwise you'd just hear a series of the opening three seconds of different tracks. I eventually managed to get some pseudocode that I thought would half-work to have a bunch of different tracks for each faction, with different keys and general moods, and switch between them without having to start over, unless you entered a new faction's space. If `Republic - Friendly, Space, No Enemies` was at 1:32.76 when a few pirate clippers jumped in, it'd fade to `Republic - Friendly, Space, Low Threat Enemies,` which would start at 1:32.76. However, when I mocked up what the result would sound like, it was invariably jarring and distracting.

For now, we'll probably have to be happy with no music implemented directly in the game. Windows Media Player will have to suffice.

**Comment by JimmyZenith** [comment]
I am working on some "Layered" style game music right now that could work for that. You start out with a full, multi-layered song and then you divide it into stems: Ambient synths and strings, Percussion, Horns and Woodwinds, Electric Guitar and Bass etc. When you're just flying around you only hear the ambient track, but with each change in game state, different layers are triggered to fade in and out. When hostiles target you, when the republic fleet arrives, and when you jump in to a system of a different faction can all trigger these. However, we need to keep in mind, that there can be LOTS of rapid fire sounds in a battle in this game. Minimalism might be the way to go with the music in this game. 
Are you guys using middleware like Wwise or FMOD? I just downloaded Wwise but I have yet to get the hang of it. I'll bet it would be pretty easy to set up a system like that in Wwise.
I'm attaching some helpful snapshots I took at GameSoundCon in L.A. back in October.

https://photos.app.goo.gl/v9dH1bYXe3zrV6Mx8
or
![IMG_20191029_121224](https://user-images.githubusercontent.com/19825400/73908658-aa0c8500-4867-11ea-9810-b36ea41b557f.jpg)
![IMG_20191029_121312](https://user-images.githubusercontent.com/19825400/73908661-aaa51b80-4867-11ea-8dca-d9c1cd9f5491.jpg)
![IMG_20191029_121641](https://user-images.githubusercontent.com/19825400/73908665-ab3db200-4867-11ea-8784-f5e7c10b1553.jpg)
![IMG_20191029_123629](https://user-images.githubusercontent.com/19825400/73908666-ab3db200-4867-11ea-9280-41a81a6c0b33.jpg)

**Comment by jafdy** [comment]
Most pre-made solutions aren't good for us as they typically require payment to use (and I suspect a closed base to integrate into). Although I think this would be an ideal solution.

**Comment by JimmyZenith** [comment]
What I mean is, the "full, multi-layered song" that we start with is my own composition that I composed for this game, without payment, to be published in Creative Commons.

**Comment by MCOfficer** [comment]
> Edit: my name on discord is Kridd I submitted a main theme concept there already.

Oof, and i almost you were MIA. If you need motivation, my sister already asked for a copy of the IP you sent me, since she can't get it out of her head ^^

While I'm not generally opposed to experiments, i think for the game it would be best to get a good title track first (which could still be revised later) before attempting a deep dive into dynamic music.

**Comment by JimmyZenith** [comment]
Sorry I haven't communicated very well. The new semester caught me by surprise. I'm currently studying Commercial Music with an emphasis in game sound. I have decided to make this game my project for my Music Production Group class. I am currently getting musicians together to record the orchestral parts for that title track. I want this to be as professional sounding as possible so I have a good portfolio piece to start off my career with. 
Thanks for telling me that! That's good to hear!

**Comment by Amazinite** [comment]
> I'm currently studying Commercial Music with an emphasis in game sound.

Sounds like just the person we need. \:)

> Sorry I haven't communicated very well. The new semester caught me by surprise.

No worries. A fair number of us, including myself, are college students, so things do tend to slow down around the starts and ends of semesters.

> I have decided to make this game my project for my Music Production Group class.

So I'm not the only one who turns Endless Sky into class projects.

> I am currently getting musicians together to record the orchestral parts for that title track. I want this to be as professional sounding as possible so I have a good portfolio piece to start off my career with.

Best of luck to you on that. Game music is something we've needed for a long time, so it's great to see someone dedicated to seeing it done.

**Comment by JimmyZenith** [comment]
> Sounds like just the person we need. :)
> Best of luck to you on that.
Thanks! I'm honored! 
I have a bunch of student musicians and an awesome school recording studio at my disposal so I have no excuse not to get this done. Haha

**Comment by AlbertNewton** [comment]
Finally, someone who won't procrastinate like I did, _and_ who actually has skill in composition of music!

**Comment by Arrow2thekn33** [comment]
I listened to the track you uploaded on discord and I think it fits the game perfectly, it's quite good.

**Comment by JimmyZenith** [comment]
Thank you! I just need to finish it up now.

Where can I find a list of all the races and their factions and a brief description of each? 
While I'm waiting for an answer on whether I can still use my school recording studio or not. I want to work on some melodic motifs for each race.

**Comment by Arrow2thekn33** [comment]
https://endless-sky.fandom.com/wiki/Factions
Here’s a list of all the factions, clicking on each one should give you a pretty good description of them.

**Comment by fuzzyTew** [comment]
I feel like this would be moved forward a lot by even a small addition to the actual code.  Like a single song, in one situation, that isn't set in stone as being right, but is actually in the game.

**Comment by Pointedstick** [comment]
I just found https://incompetech.com/music/royalty-free/music.html which has some really great CC-4.0 licensed music.

**Comment by fuzzyTew** [comment]
There's also https://funkwhale.audio and I suppose https://ccmixter.org , both of which have apis that could eventually be used.

**Comment by samrocketman** [comment]
If there's any interest in composers in this issue adding music to the game; please ping me (samrocketman) on discord where we're coordinating music composition.

We are working on music engine support in the game as well as initially adding a plugin full of freely licensed music composed specifically for endless sky.

**Comment by SomeTroglodyte** [comment]
I just had the idea to try how the [FO muzak](https://github.com/freeorion/freeorion/blob/61de7441fe8e709b5fc2d3095879f2460e2e793f/default/data/sound/artificial_intelligence_v3.ogg) would fit... At least for the opening menu it wouldn't be too bad. Pity I can't find their attribution.
